rfcs/0013-synthea-future-claims-and-automation.md

```diff
@@ -0,0 +1,130 @@
+* [Future Possibilities](#future-possibilities)
+* [Addendums](#addendums)
+
+## Motivation
+[Motivation]: #motivation
+
+For those unfamiliar with Mitre's Synthea application for generating synthetic beneficiary and claim data, before discussing motivation and a proposed solution, here is some background on Synthea and how BFD uses it. 
+
+### Background on Synthea
+
+Mitre Synthea:
+
+* [Synthea](https://github.com/synthetichealth/synthea) -- for generating synthetic beneficiary and claim data to load into BFD DB.
+
+Generating & Loading Synthea Data: 
+- BFD clones the latest version of the synthea codebase, and generates the RIF files via the `run_synthea` command, where the batch size and geography can be set, as well as a synthea.properties file, which is important, for defining numeric ranges of key claim and beneficiary properties i.e. beneficiary ID, claim group ID, claim date, etc.
+- The RIF files are copied into the BFD Server codebase, and loaded into the local database. 
+- If no errors loading locally, the RIF files are uploaded to AWS S3 and loaded into TEST, PROD SBX, PROD databses via the same ETL process with CCW RIF files. 
+- Synthetic data is now available to BFD users.
+
+* In addition to these steps, BFD follows a thorough test plan before generating or loading data into TEST, PROD SBX, or PROD environments. 
+  * [Synthea Test Plan](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/apps/bfd-model/bfd-model-rif-samples/dev/synthea-test-plan.md) 
+
+Motivation:
+With previous Synthea data releases having spanned about 3 months apart, and only 20,000 synthetic beneficiaries released, data becomes easily outdated, and the amount of data being released doesn't justify the usecase of synthetic data. So, there is inefficiency involved with producing and releasing synthetic data when done without automation. Inefficiency that in part is caused by a process full of human erorr, causing incidents, such as accidental overlap of beneficiary IDs between different synthetic data batches. Therefore, there are more than enough reasons to look into streamlining and scaling up synthetic data production to maximize the benefit to users and partners. 
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+The proposed solution is to automate the generation of synthetic data with a set batch size on a weekly basis using the latest version of the synthea codebase.
+Each quarter, the expectation is to have significant changes to the synthetic data properties from Mitre's code. So the communications to users and partners on a quarterly basis will be more in depth compared to that of on a weekly basis. The goal is to primarly generate data on a regular basis to increase the benefit and quality of the data to the user at a given point in time. 
+
+On a weekly basis, synthetic data with a future claim date will also be generated. This data will be staged until it can be loaded properly with the claim date. 
```

Comment:
This corresponds to 'automated recurring generation' in the comment above but it jumps back and forth between talking about the solution and the communications, and then back to the solution. I suggest not even discussing the communication until you are discussion the loading of the data since that is the point at which we would need to communicate.

Reply:
We need a bit more detail on the generation of the data. Here are some questions that we probably want to answer (and some possible answers but just to illustrate the point).

What low-level mechanism will be used to generate the data? (a script of some sort is my guess - we don't have to say what language the script will be written in but we should narrow it down from being any possible means of generation -- script, compiled program, something built into BFD, ansible, a new AWS service, etc -- unless you are thinking that we would use one of those things which is fine but needs to be specified)

What parameters will the generator accept? (number of benes for instance? other things)

Where will the generator be run? (locally on a workstation (I would say no). on some provisioned resource in AWS? (probably needs to be along these lines))

Where will the generator output go? (s3 probably eventually but does it get staged locally for testing first?)

Does the generator fetch and build the latest synthea?

How does the generator get the correct starting values? How are these persisted from one generation to the next?

What sort of automated validation or checking of ranges does the generator do before and/or after generating the data?